contributors/design-proposals/arbitrator.md

```diff
@@ -0,0 +1,284 @@
+* provided all applications best-effort access to all unused resources according to some target weight (one weight assigned
+  to each application, i.e. if all applications wanted to use all free resources, then they would be allowed to do so 
+  in some relative proportion)
+* If some application A is using less than its guarantee, and then if it decides to use its guarantee and there aren't 
+  enough free resources to do so, it should be able to evict tasks from some other application or applications
+  (that is/are using more than their guarantee) in order to obtain its guarantee
+
+Further, group "bigdata" apps and "service" apps into two buckets, providing each bucket (in aggregate) guaranteed 
+access to some fraction of the cluster, and best-effort access to the entire cluster with the understanding that usage 
+above the guarantee can be revoked at any time. 
+
+According to this [mesos-style.md](https://github.com/kubernetes/kubernetes/blob/master/docs/devel/mesos-style.md), 
+it definitely can build a customized components to do so; but it’s better for Kubernetes to provide such a general feature
+ about resource planning, resource management and resource sharing.
+
+## Terminology
+
+* Arbitrator: A new components/binary that allocate resources based on policy (resource allocation).
+  The default scheduler still assign Pods to resources (resource assignment) 
+* Deserved (Resource): The total number of resources that the arbitrator allocated to the namespace
+* Overused: The namespace is overused if it used more resources than deserved
+
+## Context
+
+As Kubernetes growing up, there’s several features on resource QoS and sharing between namespaces.
+
+### Admission and ResourceQuota:
+In kubernetes, namespace-level quota defines the max resources that a namespace can use. It's achieved by ResourQuotaAdmission
+ which reject pod creation if exceed the quota. If total Quota is bigger than capability of cluster, it does not define how
+ to share resources between namespace; and if one namespace use less resource than its quota and other exceed, it can not
+  borrow those resources to other and then preempt back when necessary.
+
+The admission control is a general concept: Kubernetes can reject pod creation based on an arbitrary policy. For example, 
+even if an application has enough quota to run a pod, it can reject pod creation if the application would then be using 
+a disproportionate amount of resources in the cluster. There’s discussion on doing admission control on a per-"job" 
+(really per-controller) basis, in which case an admission controller can act like a batch queueing system.
+
+As a general concept, a customized admission can be built to reject pod creation; but to achieve the following requirements,
+ an arbitrator controller was introduced:
+
+* Calculate deserved resource of each namespace based on __namespace’s request__ (e.g. pending pods)
```

Comment:
pending or just non-terminal pods?  isnt the deserved resource the minimum resource guarantee made to the namespace?  can you discuss why having a namespace scoped resource is preferred to define arbitration policies versus a cluster scoped resource that selects a set of namespaces via a matching label selector?  the tenancy and/or workload boundary may not be the namespace in all deployments.

for context, in openshift, we have a `ClusterResourceQuota` that allows a quota to target a set of namespaces via a matching selector.  i imagine arbitration policies at that boundary would also be useful.

see: https://docs.openshift.org/latest/admin_guide/multiproject_quota.html